esphome/components/mitsubishi_itp/climate.py

```diff
@@ -0,0 +1,462 @@
+    CONF_SUPPORTED_MODES,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_FREQUENCY,
+    DEVICE_CLASS_HUMIDITY,
+    ENTITY_CATEGORY_CONFIG,
+    ENTITY_CATEGORY_NONE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_CELSIUS,
+    UNIT_HERTZ,
+    UNIT_PERCENT,
+)
+from esphome.core import coroutine
+
+CODEOWNERS = ["@Sammy1Am", "@KazWolfe"]
+
+AUTO_LOAD = [
+    "climate",
+    "select",
+    "sensor",
+    "binary_sensor",
+    "button",
+    "text_sensor",
+    "time",
+]
+DEPENDENCIES = [
+    "uart",
+    "climate",
+]
+
+CONF_UART_HEATPUMP = "uart_heatpump"
+CONF_UART_THERMOSTAT = "uart_thermostat"
+CONF_TIME_SOURCE = "time_source"
+
+CONF_THERMOSTAT_TEMPERATURE = "thermostat_temperature"
+CONF_THERMOSTAT_HUMIDITY = "thermostat_humidity"
+CONF_THERMOSTAT_BATTERY = "thermostat_battery"
+CONF_ERROR_CODE = "error_code"
+CONF_ISEE_STATUS = "isee_status"
+
+CONF_SELECTS = "selects"
+CONF_TEMPERATURE_SOURCE_SELECT = "temperature_source_select"  # This is to create a Select object for selecting a source
+CONF_VANE_POSITION_SELECT = "vane_position_select"
+CONF_HORIZONTAL_VANE_POSITION_SELECT = "horizontal_vane_position_select"
+
+CONF_BUTTONS = "buttons"
+CONF_FILTER_RESET_BUTTON = "filter_reset_button"
+
+CONF_TEMPERATURE_SOURCES = (
+    "temperature_sources"  # This is for specifying additional sources
+)
+
+CONF_DISABLE_ACTIVE_MODE = "disable_active_mode"
+CONF_ENHANCED_MHK_SUPPORT = (
+    "enhanced_mhk"  # EXPERIMENTAL. Will be set to default eventually.
+)
+
+DEFAULT_POLLING_INTERVAL = "5s"
+
+mitsubishi_itp_ns = cg.esphome_ns.namespace("mitsubishi_itp")
+MitsubishiUART = mitsubishi_itp_ns.class_(
+    "MitsubishiUART", cg.PollingComponent, climate.Climate
+)
+
+TemperatureSourceSelect = mitsubishi_itp_ns.class_(
+    "TemperatureSourceSelect", select.Select
+)
+VanePositionSelect = mitsubishi_itp_ns.class_("VanePositionSelect", select.Select)
+HorizontalVanePositionSelect = mitsubishi_itp_ns.class_(
+    "HorizontalVanePositionSelect", select.Select
+)
+
+FilterResetButton = mitsubishi_itp_ns.class_(
+    "FilterResetButton", button.Button, cg.Component
+)
+
+DEFAULT_CLIMATE_MODES = ["OFF", "HEAT", "DRY", "COOL", "FAN_ONLY", "HEAT_COOL"]
+DEFAULT_FAN_MODES = ["AUTO", "QUIET", "LOW", "MEDIUM", "HIGH"]
+CUSTOM_FAN_MODES = {"VERYHIGH": mitsubishi_itp_ns.FAN_MODE_VERYHIGH}
+VANE_POSITIONS = ["Auto", "1", "2", "3", "4", "5", "Swing"]
+HORIZONTAL_VANE_POSITIONS = ["Auto", "<<", "<", "|", ">", ">>", "<>", "Swing"]
+
+INTERNAL_TEMPERATURE_SOURCE_OPTIONS = [
+    mitsubishi_itp_ns.TEMPERATURE_SOURCE_INTERNAL
+]  # These will always be available
+
+validate_custom_fan_modes = cv.enum(CUSTOM_FAN_MODES, upper=True)
+
+BASE_SCHEMA = climate.CLIMATE_SCHEMA.extend(
+    {
+        cv.GenerateID(CONF_ID): cv.declare_id(MitsubishiUART),
+        cv.Required(CONF_UART_HEATPUMP): cv.use_id(uart.UARTComponent),
+        cv.Optional(CONF_UART_THERMOSTAT): cv.use_id(uart.UARTComponent),
+        cv.Optional(CONF_TIME_SOURCE): cv.use_id(time.RealTimeClock),
```

Comment:
Please use the existing `CONF_TIME_ID` const.
```suggestion
        cv.Optional(CONF_TIME_ID): cv.use_id(time.RealTimeClock),
```

This could also be automatically set if the user has a `time` config:
```suggestion
        cv.OnlyWith(CONF_TIME_ID, "time"): cv.use_id(time.RealTimeClock),
```